### PR TITLE
Add linting workflow for ament_cmake packages

### PR DESCRIPTION
**Pull request by piraka9011**
## Changes

* Update `README`
* Add a linting workflow 

Signed-off-by: Anas Abou Allaban <allabana@amazon.com>

### CODE REVIEW

**Comment by piraka9011** [comment]
> Should this PR also include documentation on usage in the `Usage` section?

Maybe just a link to the repo should be good. I've already linked the file which is technically an example of how to use.

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Lint ROS2 package
+# Run this action whenever a pull request is made
+on:
+  pull_request:
+
+# Specify the linting job IDs to process.
```
Comment (thomas-moulard) [inline .github/workflows/lint.yml]: What does this mean?

File: README.md [hunk 2]
```diff
@@ -1 +1,11 @@
+# ROS2 GitHub Action CI Template
 
+A template repo to help ROS2 developers create a GitHub repository with the ROS2 CI GitHub Action.
```
Comment (thomas-moulard) [inline README.md]: Avoid passive voice, compare with https://github.com/actions/javascript-action

"Use this template to bootstrap the creation of a JavaScript action.🚀"

**Comment by dabonnie** [comment]
LGTM other than adding details in the README.

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+name: Lint ROS2 package
+# Run this action whenever a pull request is made
+on:
+  pull_request:
+
+# Specify the linting job IDs to process. 
+jobs:
+  ament_lint:
```
Comment (thomas-moulard) [inline .github/workflows/lint.yml]: add a name here?

File: .github/workflows/lint.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    runs-on: ubuntu-18.04
+    strategy:
+      fail-fast: false  # We still want other linters to run if one fails
+      # We specify the linters we want to use here.
+      # An up to date list can be found at https://github.com/ament/ament_lint
```
Comment (thomas-moulard) [inline .github/workflows/lint.yml]: https://github.com/ament/ament_lint/tree/eloquent

File: .github/workflows/lint.yml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+      matrix:
+        linter: [copyright, cppcheck, cpplint, lint_cmake, uncrustify, xmllint]
```
Comment (thomas-moulard) [inline .github/workflows/lint.yml]: Explain that e.g. cppcheck only apply to C++ packages?

File: .github/workflows/lint.yml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    steps:
+      - uses: actions/checkout@v1
+      - uses: ros-tooling/setup-ros2@master
+      - uses: ros-tooling/action-ros2-lint@master
```
Comment (thomas-moulard) [inline .github/workflows/lint.yml]: Never use master in workflows, people should depend on releases instead.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
+# Contributing Guidelines
+Thank you for your interest in contributing to our project. Whether it's a bug report, new feature, correction, or additional
```
Comment (thomas-moulard) [inline CONTRIBUTING.md]: Check with mdlint.

File: cpp_template_pkg/package.xml [hunk 8]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>cpp_template_pkg</name>
+  <version>0.0.0</version>
+  <description>A template packge used to demonstrate ROS2 GitHub Actions</description>
```
Comment (thomas-moulard) [inline cpp_template_pkg/package.xml]: typo

File: cpp_template_pkg/package.xml [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+  <maintainer email="ros-tooling@googlegroups.com">ROS Tooling Working Group</maintainer>
```
Comment (thomas-moulard) [inline cpp_template_pkg/package.xml]: remove and put FIXME here - we don't want to get pinged about failures of all the projects coming from this template.

File: CONTRIBUTING.md (hunk 7 above)
Comment (piraka9011) [reply CONTRIBUTING.md]: Do we want to have a sentence on each line or stick to a certain line length?

Comment (thomas-moulard) [reply CONTRIBUTING.md]: The ROS style is one sentence per line and 80 cols I think we could go with that.

File: .github/workflows/lint.yml (hunk 1 above)
Comment (piraka9011) [reply .github/workflows/lint.yml]: Specifying the job ID workflows: https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_id

Not sure if there's a better way to put it.

**Comment by thomas-moulard** [comment]
It may be me, but this package content does not match the output of "ros2 pkg create cpp_template_pkg --node-name cpp_template_pkg" ?

**Comment by piraka9011** [comment]
> It may be me, but this package content does not match the output of "ros2 pkg create cpp_template_pkg --node-name cpp_template_pkg" ?

It was made using `ros2 pkg create --package-format 3 --build-type ament_cmake`.
Although it looks like the whole repo has to be one package?

**Comment by thomas-moulard** [comment]
I see - what about adding `--node-name` so that we actually have something to build?

File: .github/workflows/lint.yml (hunk 1 above)
Comment (thomas-moulard) [reply .github/workflows/lint.yml]: I see what you mean, actually ament_lint_cpp is the job id here, not "jobs". Maybe we can just drop this comment, I'm not sure how it helps user?

**Comment by thomas-moulard** [comment]
https://github.com/errata-ai/vale + enable Google / write-good

**Comment by thomas-moulard** [comment]
Could you fix this PR? This is not doing what it's stating and incorporates a bunch of AWS logic (signing CLA, etc.)

**Comment by piraka9011** [comment]
> Could you fix this PR? This is not doing what it's stating and incorporates a bunch of AWS logic (signing CLA, etc.)

Thought we want to have things like LICENSE and Contributing to this as well. Will fix up.

**Comment by thomas-moulard** [comment]
For this PR, let's focus on adding the lint workflow.

**Comment by prajakta-gokhale** [comment]
@piraka9011 is this ready for merge?

File: cpp_template_pkg/package.xml [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+<package format="3">
+  <name>cpp_template_pkg</name>
+  <version>0.0.0</version>
+  <description>A template package used to demonstrate ROS2 GitHub Actions</description>
+  <maintainer email="foo@bar.baz">Foobar</maintainer>
+  <license>Apache 2.0</license>
+
+  <buildtool_depend>ament_cmake</buildtool_depend>
+
+  <test_depend>ament_copyright</test_depend>
```
Comment (emersonknapp) [inline cpp_template_pkg/package.xml]: Do you need copyright and common, or can you just pull auto and get all others recursively?

**Pull request merged**